routes/maintenance.py

```diff
@@ -85,6 +85,16 @@ def handle(self, app, cmd):
         elif cmd == 'ssh':
             subprocess.call(['systemctl', 'restart', 'ssh'], stderr=self.void)
             return self.jsonify({'ssh': True})
+        elif cmd == 'backup':
+            subprocess.call(['tar', '-czf', '/boot/settings.tar.gz', '/root/photoframe_config'], stderr=self.void)
+            return 'Backup Successful', 200
+        elif cmd == 'restore':
+            if os.path.isfile("/boot/settings.tar.gz"):
+                subprocess.call(['tar', '-xzf', '/boot/settings.tar.gz', '-C', '/'], stderr=self.void)
```

Comment:
This should not be done within the frame, since we're multi-threaded, instead it's better to have a tool that you kick off which will shutdown the frame service, update it and then restart it. This also gives us the option to add extra steps during import of older configurations.

Reply:
Also, the restore here does not remove the old configuration, so you run a big risk of getting a really jumbled setup after doing restore.

One way to do restore without having downtime is to restore into `/root/photoframe_config_restore` and once that is done, validate configuration and then finally shutdown, swap folders and start again. Once it successfully starts, it should delete the old setup. This also gives a safety net should something fail, allowing rollback.